assets/queries/cloudFormation/s3_buckets_are_publicly_accessible/query.rego

```diff
@@ -0,0 +1,50 @@
+package Cx
+
+CxPolicy [ result ] {
+ 	resourcePolicy := input.document[indexPolicy].Resources[namePolicy]
+	policyProperties := [policyProperties | resourcePolicy.Type == "AWS::IAM::Policy"; policyProperties = resourcePolicy.Properties.PolicyDocument]
+	policyStatements := policyProperties[_].Statement[index]
+    
+  resourceBucket := input.document[indexBucket].Resources[nameBucket]
+  bucketProperties := [bucketProperties | resourceBucket.Type == "AWS::S3::Bucket"; bucketProperties = resourceBucket.Properties]
```

Comment:
I think you have to look for "AWS::S3::BucketPolicy" policies instead of "AWS::IAM::Policy" policies. Also, they must be related to the bucket you are currently checking.

According to the query description, you have to check if there is a policy with [Effect='Allow' and (Principal='\*' or Principal.AWS='\*')]. The way you are doing, the query is checking for each policy, which means, it will return a result for each policy with [Effect='Allow' and (Principal='\*' or Principal.AWS='\*')].

Replace with:
```
resourceBucket := input.document[indexBucket].Resources[nameBucket]
resourceBucket.Type == "AWS::S3::Bucket"
    
policyStatements := [policyStatement | 	resourcePolicy := input.document[_].Resources[_]
    				        resourcePolicy.Type == "AWS::S3::BucketPolicy"
                                        checkRef(resourcePolicy.Properties.Bucket, nameBucket)
    					policyStatement := resourcePolicy.Properties.PolicyDocument.Statement[_]]
```

Reply:
Done